src/gui/qgsexpressiontreeview.cpp

```diff
@@ -258,35 +258,38 @@ void QgsExpressionTreeView::updateFunctionTree()
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "[]" ), QStringLiteral( "[ ]" ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "||" ), QStringLiteral( " || " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "BETWEEN" ), QStringLiteral( " BETWEEN " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "NOT BETWEEN" ), QStringLiteral( " NOT BETWEEN " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "IN" ), QStringLiteral( " IN " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "LIKE" ), QStringLiteral( " LIKE " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "ILIKE" ), QStringLiteral( " ILIKE " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "IS" ), QStringLiteral( " IS " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "IS NOT" ), QStringLiteral( " IS NOT " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "OR" ), QStringLiteral( " OR " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "AND" ), QStringLiteral( " AND " ) );
-  registerItem( QStringLiteral( "Operators" ), QStringLiteral( "NOT" ), QStringLiteral( " NOT " ) );
+  const QStringList operators = QStringList() << QStringLiteral( "+" )
+                                << QStringLiteral( "-" )
+                                << QStringLiteral( "*" )
+                                << QStringLiteral( "/" )
+                                << QStringLiteral( "%" )
+                                << QStringLiteral( "^" )
+                                << QStringLiteral( "=" )
+                                << QStringLiteral( "~" )
+                                << QStringLiteral( ">" )
+                                << QStringLiteral( "<" )
+                                << QStringLiteral( "<>" )
+                                << QStringLiteral( "<=" )
+                                << QStringLiteral( ">=" )
+                                << QStringLiteral( "[]" )
+                                << QStringLiteral( "||" )
+                                << QStringLiteral( "BETWEEN" )
+                                << QStringLiteral( "NOT BETWEEN" )
+                                << QStringLiteral( "IN" )
+                                << QStringLiteral( "LIKE" )
+                                << QStringLiteral( "ILIKE" )
+                                << QStringLiteral( "IS" )
+                                << QStringLiteral( "IS NOT" )
+                                << QStringLiteral( "OR" )
+                                << QStringLiteral( "AND" )
+                                << QStringLiteral( "NOT" );
+  for ( const auto &name : operators )
+  {
+    registerItem( QStringLiteral( "Operators" ), name,  ' ' + name + ' ', QString(), QgsExpressionItem::ExpressionNode, false, -1, QIcon(), QgsExpression::tags( name ) );
```

Comment:
Oh. Good catch. I will have a look at it.

Reply:
But btw. should it really have a space? With `[]` inserted it seems for me more comfortable to click between and enter the index number instead of needing to remove the space and then add the index number (or mark the space first).

Reply:
> But btw. should it really have a space? With `[]` inserted it seems for me more comfortable to click between and enter the index number instead of needing to remove the space and then add the index number (or mark the space first).

I agree.

Reply:
But still there should be no leading space....